### PR TITLE
[crux-mir-comp] struct support for munge

### PR DESCRIPTION
**Pull request by plredmond**
@spernsteiner I did not see tests for `munge` in the `crux-mir-comp/` package directory. Also, I have set the base for this PR to the `crux-mir-comp-readme` branch. Please let me know if I should merge against `master` instead.

* [x] add a test file which calls munge on a struct

### CODE REVIEW

File: crux-mir-comp/src/Mir/Cryptol.hs [hunk 1]
```diff
@@ -326,7 +326,11 @@ munge sym shp rv = do
                     W4.justPartExpr sym <$> go shp rv
                 return $ MirVector_PartialVector pv'
             MirVector_Array _ -> error $ "munge: MirVector_Array NYI"
-        -- TODO: StructShape
+        go (StructShape _ _ flds) (AnyValue tr rvs)
+            | StructRepr cr <- tr
```
Comment (spernsteiner) [inline crux-mir-comp/src/Mir/Cryptol.hs:330]: Nit: The naming convention in crux-mir is generally `tpr` for values of type `TypeRepr tp` and `ctx` for `CtxRepr ctx` (a.k.a `Assignment TypeRepr ctx`)

**Review by spernsteiner — COMMENTED** [review]
The `StructShape` case looks good.

>I have set the base for this PR to the crux-mir-comp-readme branch. Please let me know if I should merge against master instead.

Yes, please rebase this branch onto `master` and retarget the PR.  `crux-mir-comp-readme` is readme changes only, so you shouldn't run into any trouble working off `master` instead.

**Comment by plredmond** [comment]
Sorry about the force-pushes. Rebased the branch on to an old `master` at first.

**Review by spernsteiner — COMMENTED** [review]
Looks good - thanks!

If you add the "ready to merge" label, then I believe we have a bot that will merge the PR once CI passes

File: crux-mir-comp/test/symb_eval/comp/munge_struct.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+fn reflect(p: Point) -> Point {
+    Point{x: p.y, y: p.x}
+}
+
+#[crux_test]
+fn munge_struct_equiv_test() {
+    let (x, y) = <(u32, u32)>::symbolic("p");
+    let p2 = reflect(munge(Point{x, y}));
```
Comment (plredmond) [inline crux-mir-comp/test/symb_eval/comp/munge_struct.rs:18]: Oops, the munge might be better placed outside of reflect. Either way the test tests that it doesn't crash crux-mir-comp, but this way it's a little weird looking.

**Review by spernsteiner — APPROVED** [review]

File: crux-mir-comp/test/symb_eval/comp/munge_struct.rs (hunk 2 above)
Comment (spernsteiner) [reply crux-mir-comp/test/symb_eval/comp/munge_struct.rs:18]: Should work either way, just as long as the argument is a struct

Comment (plredmond) [reply crux-mir-comp/test/symb_eval/comp/munge_struct.rs:18]: Ok

**Comment by plredmond** [comment]
Just pushed the golden file for the test. I'll add the ready to merge label.

**Comment by plredmond** [comment]
I don't have perms to add labels.

**Pull request merged**